### PR TITLE
refactor: disable components when loading data

### PR DESCRIPTION
**Pull request by NullRa**

### CODE REVIEW

File: Common/MainViewController.swift [hunk 1]
```diff
@@ -59,16 +59,52 @@ class MainViewController: UIViewController, BikeAndBusDelegate, UISearchBarDeleg
...
+
+    func setMapViewControl(enable: Bool) {
+        viewModel.isMapControl = enable
+        mainMapView.isZoomEnabled = enable
+        mainMapView.isPitchEnabled = enable
+        mainMapView.isRotateEnabled = enable
+        mainMapView.isScrollEnabled = enable
+    }
+
+    func setRefresshButton(enable: Bool) {
```
Comment (ender503) [inline Common/MainViewController.swift]: Typo: `setRefreshButton`

File: Common/MainViewController.swift [hunk 2]
```diff
@@ -59,16 +59,52 @@ class MainViewController: UIViewController, BikeAndBusDelegate, UISearchBarDeleg
...
+            showAlertMessage(title: "定位權限已關閉",
+                             message: "如要變更權限，請至 設定 > 隱私權 > 定位服務 開啟",
+                             actionTitle: "確認")
+            return
+        }
+        if CLLocationManager.authorizationStatus() == .notDetermined {
+            MapManager.shared.manager.requestAlwaysAuthorization()
+        }
+        MapManager.shared.manager.startUpdatingLocation()
+    }
```
Comment (ender503) [inline Common/MainViewController.swift]: Is this function called by someone? 🤔 
It looks like a little bit lonely. 😏

File: Common/MainViewController.swift [hunk 3]
```diff
@@ -59,16 +59,52 @@ class MainViewController: UIViewController, BikeAndBusDelegate, UISearchBarDeleg
...
 
         DispatchQueue.global().async {
             self.viewModel.updateUbikeData()
             DispatchQueue.main.async {
                 self.updateAnnotations(annotations: self.viewModel.getRegionStations())
                 self.loadingView.stopAnimating()
-                self.refreshButton.isEnabled = true
+                self.setRefresshButton(enable: true)
+                self.setLocationButton(enable: true)
+                self.setMapViewControl(enable: true)
```
Comment (ender503) [inline Common/MainViewController.swift]: You are trying to disable these components when fetching the station data, so you could merge these function into `setLoading()` or `setFetching()`, and use it to set all the components enable or not.

File: Common/MainViewController.swift (hunk 2 above)
Comment (ender503) [reply Common/MainViewController.swift]: Too lonely ...

File: Common/MainViewController.swift (hunk 1 above)
Comment (NullRa) [reply Common/MainViewController.swift]: 😱

File: Common/MainViewController.swift (hunk 2 above)
Comment (NullRa) [reply Common/MainViewController.swift]: My fault, I think it was removed early. 🤥

File: Common/MainViewController.swift (hunk 3 above)
Comment (NullRa) [reply Common/MainViewController.swift]: OK 👍

File: Common/MainMapViewModel.swift [hunk 4]
```diff
@@ -9,7 +9,9 @@ class MainMapViewModel {
     var center: CLLocationCoordinate2D?
     var isAutoUpdate = true
     var isSearchBarCollapsed = true
+    var isMapControl = true
     var isLocation = true
+    var isRefresh = true
```
Comment (ender503) [inline Common/MainMapViewModel.swift]: `isMapControl`, `isLocation` and `isRefresh` are set, but not be used by anyone.
It would be nice to remove them. 😄

Comment (NullRa) [reply Common/MainMapViewModel.swift]: So, This is all unnecessary. 😱
If it wouldn't be used, don't need to set it value.

螢幕快照 2019-07-30 上午8 13 44

**Comment by NullRa** [comment]
Should these functions be removed, if it hadn't be used in ViewModel?

螢幕快照 2019-07-30 上午8 22 17


And, In the ViewController, all of the functions setting true/false gather in one function, right?

螢幕快照 2019-07-30 上午8 23 06

I fixed it, review, please.
Thank you. 😚
If it's okay, I will remove the commands.

**Pull request merged**